powersimdata/data_access/data_access.py

```diff
@@ -70,17 +70,23 @@ def write(self, filepath, data):
 
         print("Writing %s" % filepath)
 
-        with self.local_fs.openbin(filepath, mode="w") as f:
-            if ext == "pkl":
-                pickle.dump(data, f)
-            elif ext == "csv":
-                data.to_csv(f)
-            elif ext == "mat":
-                savemat(f, data, appendmat=False)
-            else:
-                raise ValueError("Unknown extension! %s" % ext)
+        if save_local:
+            f = self.local_fs.openbin(filepath, mode="w")
+        else:
+            f = self.fs.openbin(filepath, mode="w")
```

Comment:
I feel like the `if/else` could be built into the inheritance (or some other) structure, but not sure how it would look. I'll see if I can come up with something unless you get to it first.

Reply:
I agree; I don't think this is the cleanest solution, but I also think it might be a little premature to try and design it now as I'm imagining a bigger change that would be outside the scope of this PR. I'm still not sure that having the local filesystem as part of the data access class makes the most sense since I'm envisioning a scenario object with multiple data access objects (e.g. SSH, blob, local).

But I would love to see what you're talking about!

Reply:
What about something like this? Regarding a scenario having multiple data access objects, I'm starting to feel we can eliminate `LocalDataAccess` pending a few more changes, and combine both blob and ssh filesystems into a `RemoteDataAccess`, or even just the base class and exposing some configurability. We should discuss more, but I think it is separate from this PR.

```
    def write(self, filepath, data, save_local=True):
        self._check_file_exists(filepath, should_exist=False)
        print("Writing %s" % filepath)

        self._write(self.fs, filepath, data)
        if save_local:
            self._write(self.local_fs, filepath, data)


    def _write(self, fs, filepath, data):
        ext = os.path.basename(filepath).split(".")[-1]

        with fs.openbin(filepath) as f:
            if ext == "pkl":
                pickle.dump(data, f)
            elif ext == "csv":
                data.to_csv(f)
            elif ext == "mat":
                savemat(f, data, appendmat=False)
            else:
                raise ValueError("Unknown extension! %s" % ext)
```